Userland/DevTools/HackStudio/Git/GitRepo.cpp

```diff
@@ -1,131 +1,130 @@
 /*
  * Copyright (c) 2020, Itamar S. <itamar8910@gmail.com>
+ * Copyright (c) 2021, Conor Byrne <cbyrneee@protonmail.com>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
 #include "GitRepo.h"
 #include <LibCore/Command.h>
-#include <stdio.h>
-#include <stdlib.h>
 
 namespace HackStudio {
 
-GitRepo::CreateResult GitRepo::try_to_create(const LexicalPath& repository_root)
-{
-    if (!git_is_installed()) {
-        return { CreateResult::Type::GitProgramNotFound, nullptr };
-    }
-    if (!git_repo_exists(repository_root)) {
-        return { CreateResult::Type::NoGitRepo, nullptr };
-    }
+static GitRepo* s_the;
 
-    return { CreateResult::Type::Success, adopt_ref(*new GitRepo(repository_root)) };
+GitRepo& GitRepo::the()
+{
+    VERIFY(s_the);
+    return *s_the;
 }
 
-RefPtr<GitRepo> GitRepo::initialize_repository(const LexicalPath& repository_root)
+void GitRepo::initialize(LexicalPath root)
```

Comment:
This isn't really related to this PR, but the `GitRepo` class uses `LexicalPath` to pass paths around in many places.
We should refactor the code to use `String`s instead, `LexicalPath` is a 'heavier' object that's meant for path parsing and validation.
But of course this doesn't have to be fixed in this PR.